imageio/plugins/pillow.py

```diff
@@ -288,31 +273,54 @@ def write(self, image, *, mode="RGB", format=None, **kwargs):
 
         """
 
-        # ensure that the image has (at least) one batch dimension
-        if image.ndim == 3 and _is_multichannel(mode):
-            image = image[None, ...]
-            save_all = False
-        elif image.ndim == 2 and not _is_multichannel(mode):
-            image = image[None, ...]
-            save_all = False
+        if mode is None:
+            # there is no explicit function in pillow to determine the mode.
+            # However, pillow does it in Image.fromarray, and we imitate that
+            # here
+
+            # pillow supports 1 - 4 color channels depending on the format
+            # and all are channel-last
+            if image.shape[-1] in [2, 3, 4]:
```

Comment:
Prevent this `if` from being triggered on very narrow greyscale images:
```suggestion
            if image.ndim ==3 and image.shape[-1] in [2, 3, 4]:
```